.test-infra/metrics/sync/jenkins/syncjenkins.py

```diff
@@ -0,0 +1,211 @@
+  durFields = ('blockedDurationMillis,buildableDurationMillis,'
+               'buildingDurationMillis,executingTimeMillis,queuingDurationMillis,'
+               'totalDurationMillis,waitingDurationMillis')
+  fields = (f'result,timestamp,id,url,builtOn,building,duration,'
+            f'estimatedDuration,fullDisplayName,actions[{durFields}]')
+  url = f'{jobUrl}api/json?depth=1&tree=builds[{fields}]'
+  r = requests.get(url)
+  return r.json()[u'builds']
+
+
+def buildRowValuesArray(jobName, build):
+  timings = next((x
+                  for x in build[u'actions']
+                  if (u'_class' in x)
+                    and (x[u'_class'] == u'jenkins.metrics.impl.TimeInQueueAction')),
+                 None)
+  values = [jobName,
+          int(build[u'id']),
+          build[u'url'],
+          build[u'result'],
+          datetime.fromtimestamp(build[u'timestamp'] / 1000),
+          build[u'builtOn'],
+          build[u'duration'],
+          build[u'estimatedDuration'],
+          build[u'fullDisplayName'],
+          timings[u'blockedDurationMillis'] if timings is not None else -1,
+          timings[u'buildableDurationMillis'] if timings is not None else -1,
+          timings[u'buildingDurationMillis'] if timings is not None else -1,
+          timings[u'executingTimeMillis'] if timings is not None else -1,
+          timings[u'queuingDurationMillis'] if timings is not None else -1,
+          timings[u'totalDurationMillis'] if timings is not None else -1,
+          timings[u'waitingDurationMillis'] if timings is not None else -1]
+  return values
+
+
+def insertRow(cursor, rowValues):
+  cursor.execute(f'insert into {jenkinsBuildsTableName} values (%s, %s, %s, %s,'
+                  '%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)', rowValues)
+
+
+def fetchNewData():
+  connection = initConnection()
+  cursor = connection.cursor()
+  syncedJobs = fetchSyncedJobsBuildVersions(cursor)
+  cursor.close()
+  connection.close()
+
+  newJobs = fetchJobs()
+
+  for newJobName, newJobLastBuildId, newJobUrl in newJobs:
+    syncedJobId = syncedJobs[newJobName] if newJobName in syncedJobs else -1
+    if newJobLastBuildId > syncedJobId:
+      builds = fetchBuildsForJob(newJobUrl)
+      builds = [x for x in builds if int(x[u'id']) > syncedJobId]
+
+      connection = initConnection()
+      cursor = connection.cursor()
+
+      for build in builds:
+        if build[u'building']:
+          continue;
+        rowValues = buildRowValuesArray(newJobName, build)
+        print("inserting", newJobName, build[u'id'])
+        insertRow(cursor, rowValues)
+
+      cursor.close()
+      connection.commit()
+      connection.close()  # For some reason .commit() doesn't push data
+
+def probeJenkinsIsUp():
+  sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+  result = sock.connect_ex(('builds.apache.org', 443))
+  return True if result == 0 else False
+
+
+################################################################################
```

Comment:
Can you add an `if __name__ == '__main__'` to gate the execution of this script? That way anyone can add unit tests if necessary.